### PR TITLE
doc/admin/external_service/github.md: Add note on enterprise rate limiting

### PR DESCRIPTION
**Pull request by indradhanush**
## Test plan

Not applicable. Doc changes only.

### CODE REVIEW

**Review by sashaostrikov — APPROVED** [review]
LGTM!

File: doc/admin/external_service/github.md [hunk 1]
```diff
@@ -68,6 +68,18 @@ No [token scopes](https://docs.github.com/en/developers/apps/building-oauth-apps
 
 You should always include a token in a configuration for a GitHub.com URL to avoid being denied service by GitHub's [unauthenticated rate limits](https://developer.github.com/v3/#rate-limiting). If you don't want to automatically synchronize repositories from the account associated with your personal access token, you can create a token without a [`repo` scope](https://developer.github.com/apps/building-oauth-apps/scopes-for-oauth-apps/#available-scopes) for the purposes of bypassing rate limit restrictions only.
 
+## GitHub enterprise rate limits
```
Comment (unknwon) [inline doc/admin/external_service/github.md:71]: nit: GitHub offers "enterprise" both on GitHub Enterprise Server and GitHub Enterprise Cloud, based on my reading, here should be explicit about this is for GitHub Enterprise Server.

Comment (indradhanush) [reply doc/admin/external_service/github.md:71]: Good point. I just checked and it seems like Enterprise Cloud[1] and Enterprise Server[2] both offer this API. I'll create a follow up PR to mention this. 

1: https://docs.github.com/en/enterprise-cloud@latest/rest/rate-limit
2: https://docs.github.com/en/enterprise-server@3.4/rest/rate-limit

Comment (unknwon) [reply doc/admin/external_service/github.md:71]: TIL 👍 Are they also sharing the same `curl` URL?

Comment (indradhanush) [reply doc/admin/external_service/github.md:71]: Good question. I looked again and they don't share the same URL. Also in the process I found out that API rate limit is always enabled for Cloud, so this section should only be relevant for Server, like you suggested initially above. :sweat_smile: 

Coming up with a follow up PR in 2 mins. :hourglass_flowing_sand:

Comment (indradhanush) [reply doc/admin/external_service/github.md:71]: Follow up here: https://github.com/sourcegraph/sourcegraph/pull/34404

**Pull request merged**